### PR TITLE
*: remove useless conversions and omit some types

### PR DESCRIPTION
**Pull request by Rustin-Liu**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve?

Problem Summary:

When I reading the source code, I found many useless conversions. So I try to remove it.

### What is changed and how it works?

What's Changed:

remove useless conversion and omit some types.

How it Works:

### Related changes

None

### Check List 

Tests 

None

Side effects

None

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@ngaut  Pass all tests, PTAL~

File: distsql/distsql.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func GetSystemEndian() tipb.Endian {
 }
 
 func init() {
-	var i int = 0x0100
+	var i = 0x0100
```
Comment (ngaut) [inline distsql/distsql.go:191]: i := 0x0100

File: expression/builtin_compare_vec.go [hunk 2]
```diff
@@ -725,8 +725,8 @@ func (b *builtinLeastTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Co
 	}
 	var argTime types.Time
 
-	var findInvalidTime []bool = make([]bool, n)
-	var invalidValue []string = make([]string, n)
+	var findInvalidTime = make([]bool, n)
```
Comment (ngaut) [inline expression/builtin_compare_vec.go:728]: Use := instead of var definition.

File: distsql/distsql.go (hunk 1 above)
Comment (Rustin-Liu) [reply distsql/distsql.go]: Fixed.

File: expression/builtin_compare_vec.go (hunk 2 above)
Comment (Rustin-Liu) [reply expression/builtin_compare_vec.go]: Fixed

**Comment by Rustin-Liu** [comment]
@sre-bot  /run-all-tests

**Comment by zz-jason** [comment]
Could you add a static check in the `Makefile` to avoid this kind of code produced again in the future?

**Comment by Rustin-Liu** [comment]
> LGTM, Good job, Are you using any lint tools?

I use the goland, so I can inspection all redundant type conversion.

@zz-jason  Maybe we can use such tool: https://github.com/mdempsky/unconvert , I will send a issue to tracking this dev enhance. And I will try to find a solution to enhance our make dev check.

**Pull request merged**